### PR TITLE
Kristianeschenburg grassmann belongs

### PR DESCRIPTION
**Pull request by nguigs**
This PR adds tools for the Grassmann Manifold, following up on #572:
- remove hard coded dimension
- add belongs, is_tangent, to_tangent methods
- change exp and log to take and return tangent vectors instead of skew-sym matrices
- add random_uniform method
- change the metric to inherit MatricesMetric so that inner products and norms can be computed with the Frobenius dot prod.

@opeltre

### CODE REVIEW

**Comment by opeltre** [comment]
Great commits! 
They need some documentation as the Grassmann representation is not unequivocal 
I managed to get sphinx working locally and can write a few docstring lines for is_tangent and to_tangent once the PR is merged

File: geomstats/_backend/tensorflow/__init__.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
     asin as arcsin,
     atan2 as arctan2,
     clip_by_value as clip,
-    concat as concatenate,
+    concat,
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:17]: Do you want to just delete it maybe?

File: geomstats/geometry/grassmannian.py [hunk 2]
```diff
@@ -50,30 +50,174 @@ def __init__(self, n, k):
...
+    def random_uniform(self, n_samples=1):
+        """Sample random points from a uniform distribution.
+
+        Following [Chikuse03]_, :math: `n_samples * n * k`scalars are sampled
+        from a standard normal distribution and reshaped to matrices,
+        the projectors on their first k columns follow a uniform distribution.
+
+        Parameters
+        ----------
+        n_samples: int,
```
Comment (ninamiolane) [inline geomstats/geometry/grassmannian.py:62]: NIT: space before : , no comma after int, "." at the end of parameters' description

File: geomstats/geometry/grassmannian.py [hunk 3]
```diff
@@ -50,30 +50,174 @@ def __init__(self, n, k):
...
+        skew = Matrices.to_skew_symmetric(vector)
+        return Matrices.bracket(base_point, skew)
+
+    @staticmethod
+    def _check_square(point):
+        """Check if a point is square.
+
+        Parameters
+        ----------
+        point
```
Comment (ninamiolane) [inline geomstats/geometry/grassmannian.py:167]: Docstring format (although we are less strict with private methods' doc)

File: tests/test_grassmannian.py [hunk 4]
```diff
@@ -40,31 +41,78 @@ def setUp(self):
...
+            pi_2 * gs.array([r_y, r_z]), gs.array([p_xy, p_yz]))
         result = self.metric.exp(
-            pi_2 * gs.array([r_y, r_z]),
-            gs.array([p_xy, p_yz]))
+            vec, gs.array([p_xy, p_yz]))
         expected = gs.array([p_yz, p_xz])
         self.assertAllClose(result, expected)
 
-    @geomstats.tests.np_only
+    @geomstats.tests.np_and_tf_only
```
Comment (ninamiolane) [inline tests/test_grassmannian.py:57]: Nice!

File: geomstats/_backend/tensorflow/__init__.py (hunk 1 above)
Comment (nguigs) [reply geomstats/_backend/tensorflow/__init__.py:17]: No, I just needed to pass the default `axis=0` to match numpy's behaviour

**Pull request merged**